src/test/java/teammates/test/cases/webapi/BinCourseActionTest.java

```diff
@@ -0,0 +1,116 @@
+package teammates.test.cases.webapi;
+
+import java.util.List;
+
+import org.apache.http.HttpStatus;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.CourseAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.util.Const;
+import teammates.logic.core.CoursesLogic;
+import teammates.ui.webapi.action.BinCourseAction;
+import teammates.ui.webapi.action.JsonResult;
+import teammates.ui.webapi.output.MessageOutput;
+
+/**
+ * SUT: {@link BinCourseAction}.
+ */
+public class BinCourseActionTest extends BaseActionTest<BinCourseAction> {
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.BIN_COURSE;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return PUT;
+    }
+
+    @Override
+    @Test
+    protected void testExecute() throws Exception {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        String instructorId = instructor1OfCourse1.googleId;
+
+        loginAsInstructor(instructorId);
+
+        ______TS("Not enough parameters");
+
+        verifyHttpParameterFailure();
+
+        ______TS("Typical case, 2 courses");
```

Comment:
`2 courses`??

Reply:
It looks like this means: 
2 courses associated with instructor, bin 1 of them.
We expect that the course that was not binned is still in the course list.

Reply:
It is unlikely that binning one course will cause another course to be binned. It's ok to keep the test here. Just need to make the description clear and **add** a direct verification.